### PR TITLE
Path generation for ygot.OrderedMap nodes

### PR DESCRIPTION
**Pull request by wenovus**
Sample name: `Root().RoutingPolicy().PolicyDefinition().StatementAll()`

example helper function signature:
```go
func (n *RoutingPolicy_PolicyDefinition_StatementPath) Config() ygnmi.ConfigQuery[*oc.RoutingPolicy_PolicyDefinition_Statement_OrderedMap]
```

### CODE REVIEW

File: ygnmi/types.go [hunk 1]
```diff
@@ -49,7 +49,7 @@ func NewLeafSingletonQuery[T any](parentDir string, state, scalar bool, ps PathS
 }
 
 // NewNonLeafSingletonQuery creates a new NonLeafSingletonQuery object.
-func NewNonLeafSingletonQuery[T ygot.ValidatedGoStruct](dir string, state bool, ps PathStruct, subPaths []PathStruct, schemaFn func() *ytypes.Schema) *NonLeafSingletonQuery[T] {
```
Comment (DanG100) [inline ygnmi/types.go:52]: why this change? i like the type safety of enforcing that non-leafs are go structs, no some random struct

Comment (wenovus) [reply ygnmi/types.go:52]: The reason is that this returns an ordered map which is not a GoStruct but an OrderedMap interface type. GoStructs are meant to represent a container type or a list element type, and was not meant to represent a collection of elements, so it can't really be re-used for that.

Comment (wenovus) [reply ygnmi/types.go:52]: Also I believe Go generics' type constraints doesn't allow for union types between two interface types. You can't create a type that's an OR between two interfaces.

Comment (wenovus) [reply ygnmi/types.go]: leaf/non-leaf queries have now been merged into a single one: https://github.com/openconfig/ygnmi/pull/99

As discussed this will be backwards-incompatible but will unlikely be disruptive.

**Review by DanG100 — APPROVED** [review]
